src/DynamoUtilities/ObservableHashSet.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Specialized;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Dynamo.Utilities
+{
+    internal class ObservableHashSet<T> : INotifyCollectionChanged
+    {
+        private readonly HashSet<T> set = new HashSet<T>();
+
+        public event NotifyCollectionChangedEventHandler CollectionChanged;
+
+        public void Add(T item)
+        {
+            set.Add(item);
+            CollectionChanged?.Invoke(this, new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Reset));
```

Comment:
Please use the proper notification types. You are not using them now...but someone might base their code on these types
```
    //
    // Summary:
    //     Describes the action that caused a System.Collections.Specialized.INotifyCollectionChanged.CollectionChanged
    //     event.
    public enum NotifyCollectionChangedAction
    {
        //
        // Summary:
        //     An item was added to the collection.
        Add,
        //
        // Summary:
        //     An item was removed from the collection.
        Remove,
        //
        // Summary:
        //     An item was replaced in the collection.
        Replace,
        //
        // Summary:
        //     An item was moved within the collection.
        Move,
        //
        // Summary:
        //     The contents of the collection changed dramatically.
        Reset
```

Reply:
@pinzart90 this constructor for the `NotifyCollectionChangedEventArgs` requires it to be set to `Reset`. If not, it throws an exception. There are other constructors that allow passing other enum values along with the item that has changed, but since I'm not using that information, I'm using the simplest constructor.
https://docs.microsoft.com/en-us/dotnet/api/system.collections.specialized.notifycollectionchangedeventargs.-ctor?view=net-6.0#system-collections-specialized-notifycollectionchangedeventargs-ctor(system-collections-specialized-notifycollectionchangedaction)